src/gradualizer_db.hrl

```diff
@@ -0,0 +1,33 @@
+-define(BUILT_IN_SPECS, #{
```

Comment:
why not use `gradualizer_prelude.erl` so they can be in plain string format instead of AST?

Reply:
I tried that initially but got errors, e.g. spec for undefined function `erlang:'+'`.

Reply:
I think we should fix the error you got. We should be able to trick gradualizer into thinking that the function `erlang:'+'` exists.
I think it would be a good idea to have these specs written out using their surface level syntax rather than using the internal representation as it is now. It would help  with readability quite a lot!

Reply:
I'll try to do so if we go for this solution!

Reply:
> We should be able to trick gradualizer into thinking that the function `erlang:'+'` exists.

I think it is a compilation error though. Is it? `gradualizer_prelude` is compiled, although it doesn't contain any code.

Reply:
it's a vicious circle: we made `gradualizer_prelude` into a compiled erlang module in order to be included in the escript. Although we wouldn't need it to be compiled, just parse it. As a next level of workaround what if we would introduce two new attributes `-gradualizer_spec` and `gradualizer_type` (or similar with shorter names), which would behave exactly as type and spec but without the compiler checking them. So we could include things in there which pass parsing but not the compiler. Initially this would only be used in the preludes.

Reply:
> I think it is a compilation error though. Is it? `gradualizer_prelude` is compiled, although it doesn't contain any code.

When I tried adding the typespec below to `gradualizer_prelude` I didn't get any compiler errors. So it seems to me that it should be a feasible approach. @Zalastax, can you confirm based on your experience?
```
-spec erlang:'+'(integer(), integer()) ->
			integer();
		(float(), float()) ->
			float();
		(float(), integer()) ->
			float();
		(integer(), float()) ->
			float().
```

Reply:
I believed I got a compilation error for exactly that spec (when running the test suite). I'll double check when I have the time.

Reply:
I do not get compile time errors with this spec and Erlang 19.3, 20.2, or 21.1.

Reply:
Also confirming the spec compiles with 24.2.1.